### PR TITLE
Refactor image inputs and update to new input object

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

Part of #961 

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/api/data.rst [hunk 1]
```diff
@@ -99,7 +99,7 @@ ___________________________
     ~flash.core.data.io.input.MockDataset
     ~flash.core.data.io.input.NumpyInput
     ~flash.core.data.io.input.PathsInput
-    ~flash.core.data.io.input.SequenceInput
+    ~flash.core.data.io.input.ClassificationInput
```
Comment (ethanwharris) [inline docs/source/api/data.rst:102]:
```suggestion
    ~flash.core.data.io.input.SequenceInput
```

File: flash/core/data/io/input.py [hunk 2]
```diff
@@ -374,7 +380,7 @@ class SequenceInput(
     Generic[SEQUENCE_DATA_TYPE],
     Input[Tuple[Sequence[SEQUENCE_DATA_TYPE], Optional[Sequence]]],
 ):
-    """The ``SequenceInput`` implements default behaviours for data sources which expect the input to
+    """The ``ClassificationInput`` implements default behaviours for data sources which expect the input to
```
Comment (ethanwharris) [inline flash/core/data/io/input.py:383]:
```suggestion
    """The ``SequenceInput`` implements default behaviours for data sources which expect the input to
```

File: flash/core/data/io/input.py [hunk 3]
```diff
@@ -622,7 +628,7 @@ def load_sample(self, sample: Dict[str, Any], dataset: Optional[Any] = None) ->
 
 
 class TensorInput(SequenceInput[torch.Tensor]):
-    """The ``TensorInput`` is a ``SequenceInput`` which expects the input to
+    """The ``TensorInput`` is a ``ClassificationInput`` which expects the input to
```
Comment (ethanwharris) [inline flash/core/data/io/input.py:631]:
```suggestion
    """The ``TensorInput`` is a ``SequenceInput`` which expects the input to
```

File: flash/core/data/io/input.py [hunk 4]
```diff
@@ -637,7 +643,7 @@ def load_data(
 
 
 class NumpyInput(SequenceInput[np.ndarray]):
-    """The ``NumpyInput`` is a ``SequenceInput`` which expects the input to
+    """The ``NumpyInput`` is a ``ClassificationInput`` which expects the input to
```
Comment (ethanwharris) [inline flash/core/data/io/input.py:646]:
```suggestion
    """The ``NumpyInput`` is a ``SequenceInput`` which expects the input to
```

File: flash/core/data/utilities/data_frame.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    Args:
+        file: The CSV file to read.
+
+    Returns:
+        A ``DataFrame`` containing the contents of the file.
+    """
+    try:
+        return pd.read_csv(file, encoding="utf-8")
+    except UnicodeDecodeError:
+        warnings.warn("A UnicodeDecodeError was raised when reading the CSV. This error will be ignored.")
```
Comment (tchaton) [inline flash/core/data/utilities/data_frame.py:37]: rank_zero_warn.

File: flash/core/data/utilities/paths.py [hunk 6]
```diff
@@ -75,18 +70,20 @@ def is_valid_file(x: str) -> bool:
...
-                path = os.path.join(root, fname)
-                if is_valid_file(path):
-                    item = path, class_index
-                    instances.append(item)
-    return instances
+    subdirs = list_subdirs(directory)
+    if len(subdirs) > 0:
+        for target_class in list_subdirs(directory):
+            target_dir = os.path.join(directory, target_class)
+            if not os.path.isdir(target_dir):
```
Comment (tchaton) [inline flash/core/data/utilities/paths.py:77]:
```suggestion
    subdirs = list_subdirs(directory)
    if len(subdirs) > 0:
        for target_class in subdirs:
            target_dir = os.path.join(directory, target_class)
            if not os.path.isdir(target_dir):
```

**Review by tchaton — APPROVED** [review]
LGTM ! Awesome work !

File: flash/core/data/utilities/paths.py (hunk 6 above)
Comment (ethanwharris) [reply flash/core/data/utilities/paths.py:77]: Ooh, good catch!!!

**Review by ananyahjha93 — APPROVED** [review]

**Pull request merged**